ftp/ftp.go

```diff
@@ -33,109 +30,117 @@ func Client() (*FtpClient, error) {
 	return client, nil
 }
 
-func Connect() (*FtpClient, error) {
-	config := config.FtpConfig()
+func Connect(URL string, userName string, passWord string) (*goftp.Client, error) {
 
 	ftpConfig := goftp.Config{
-		User:               config.FtpUsername,
-		Password:           config.FtpPassword,
+		User:               userName,
+		Password:           passWord,
 		ConnectionsPerHost: 10,
 		Timeout:            10 * time.Second,
 		Logger:             nil,
 	}
 
-	client, err := goftp.DialConfig(ftpConfig, config.FtpServerURL)
+	client, err := goftp.DialConfig(ftpConfig, URL)
 	if err != nil {
 		return nil, err
 	}
 
-	return &FtpClient{Client: client, FtpConfiguration: &config}, nil
+	return client, nil
 }
 
-func Pull(client *FtpClient, files []string) ([]string, error){
-	filePaths := make([]string, 0)
+func Pull(client *goftp.Client, files []string) ([]string, error) {
+	filesPulled := make([]string, 0)
 	workingDir, err := os.Getwd()
 
 	if err != nil {
 		return nil, err
 	}
 
-    for _, file := range files {
-        log.Printf("Pulling file: %s", file)
-        
-        buffer, err := os.Create(file)
-        if err != nil {
-            log.Error().Msgf("Failed to create file with error => %s", err)
-            return nil, err
-        }
-        defer buffer.Close()
-
-        path := fmt.Sprintf("%s/%s", client.FtpServerPath, file)
-
-        if  err = client.Retrieve(path, buffer); err != nil {
-            log.Error().Msgf("Failed to retreive file with error => %s", err)
-            continue
-        }
-
-		file := workingDir + "/" + file
-		if filePath, err := Decompress(file); err != nil {
-            log.Error().Msgf("Failed to decompress file on path - %s with error => %s", file, err)
-        } else {
-			filePaths = append(filePaths, filePath)
+	for _, remote_file := range files {
+		local_file := fmt.Sprintf("%s/%s", workingDir, filepath.Base(remote_file))
+		log.Info().Msgf("Pulling file '%s' from remote to %s", remote_file, local_file)
+
+		buffer, err := os.Create(local_file)
+		if err != nil {
+			log.Error().Msgf("Failed to create file with error => %s", err)
+			continue
+		}
+		defer buffer.Close()
+
+		if err = client.Retrieve(remote_file, buffer); err != nil {
+			log.Error().Msgf("Failed to retreive file with error => %s", err)
+			continue
 		}
-    } 
+		filesPulled = append(filesPulled, local_file)
+
+		// // TODO: Move to seperate function becouse decopress not always needed when pulling e.g. voice station
+		// local_file := workingDir + "/" + file
```

Comment:
remove commented code

Reply:
Are we going to separate this function into standalone operation only for images?

Reply:
we will have to do WithDecompression or not